src/resolve/resolved-dns-cache.c

```diff
@@ -623,9 +623,13 @@ static int dns_cache_put_negative(
                 calculate_until_valid(soa, dns_answer_min_ttl(answer), nsec_ttl, timestamp, true);
 
         if (i->type == DNS_CACHE_NXDOMAIN) {
+                const char *name = dns_resource_key_name(key);
+                if (dns_name_endswith(name, "resolver.arpa"))
```

Comment:
the call may fail. it's acceptable to ignore the failure, but if so, make this explicit by doing a `> 0` or `!= 0` check (to clarify how you intend to handle failures: as true or as false).

(here i#d just propagate the error though)